source/common/secret/secret_manager_impl.cc

```diff
@@ -79,5 +84,101 @@ SecretManagerImpl::findOrCreateCertificateValidationContextProvider(
                                                     secret_provider_context);
 }
 
+// We clear private key and password to avoid information leaking.
+// TODO(incfly): switch to more generic scrubbing mechanism once
+// https://github.com/envoyproxy/envoy/issues/4757 is resolved.
+void redactSecret(::envoy::api::v2::auth::Secret* secret) {
+  if (secret && secret->type_case() == envoy::api::v2::auth::Secret::TypeCase::kTlsCertificate) {
+    auto tls_certificate = secret->mutable_tls_certificate();
+    if (tls_certificate->has_private_key()) {
+      tls_certificate->mutable_private_key()->set_inline_string("[redacted]");
```

Comment:
Don't redact when this is pointing to a file?

Reply:
done.

Reply:
test case?

Reply:
Done.